### PR TITLE
add tests for setup_discount.py, resolve #8 conflicts, autoformatting

### PR DESCRIPTION
**Pull request by tnmygrwl**
ignore auto-formatting changes in `setup_discount.py`

### CODE REVIEW

File: ada_tools/requirements.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@ pygeos>=0.8.0             # added manually
 pyparsing>=2.4.7          # via ada_tools (setup.py), snuggs
 pyproj>=2.6.1.post1       # via ada_tools (setup.py), geopandas
 python-dateutil>=2.8.1    # via ada_tools (setup.py), pandas
+pytest-mock>=3.11.1
```
Comment (ginic) [inline ada_tools/requirements.txt]: I think you should also list out the testing dependencies in the `ada_tools/setup.py` instead, just like the dev dependencies are listed separately. It's generally a good practice to call out test dependencies separately for packaging purposes and we should try to use the same conventions that the Red Cross is using.

**Review by ginic — CHANGES_REQUESTED** [review]
The tests themselves look good and pass for me! 
I think there are some packaging nuances to take care of before this can be merged in. Besides moving the test dependencies from `requirements.txt` to `setup.py`, I think the unit tests should live under `ada_tools/tests`, instead of `ada_tools/src/tests` in order to keep the test code separate from application code. This avoids some potential issues with importing or packaging tests unnecessarily. You can see more details on that at https://pytest.org/en/7.4.x/explanation/goodpractices.html#tests-outside-application-code. I think you can just move the tests directory and everything will work as expected. 
Finally, you may want to add a note to the ada_tools README on how to run tests.

**Comment by tnmygrwl** [comment]
thanks for pointing it out @ginic, will update the PR with another small test along with it. I will make it into draft PR for now.

File: ada_tools/requirements.txt (hunk 1 above)
Comment (tnmygrwl) [reply ada_tools/requirements.txt]: 064a452c9acd1c5a012c06dc59755c740db29ed6

**Comment by tnmygrwl** [comment]
@ginic this is ready for review. 

# Overview

Changes you've requested:
- e73e919bf571261f566fc6fd1755b2efa299bde0: removes the test from `/src/`
- 064a452c9acd1c5a012c06dc59755c740db29ed6: removes the dependency from `requirements.txt`
- 8402e569e06ab94aaa40ca765cce5a56a7620ec8: Adds another test, adds testing notes to the readme and modifies setup.py

Additional changes:
- 5492df8ce7e1435532d724dcb480284c5668f375: formatting across the package
- 29a0ef8b834f6edbe7012d70ecd7783a0780222c: resolving conflicts induced from #8 

lmk if there are test failures. I have commented the `test_main` out because testing Clicker CLI with fileops side effects is complicated and I'll get to it soon. ref: https://click.palletsprojects.com/en/8.1.x/testing/

**Review by ginic — APPROVED** [review]
Awesome! Looks great to me! Thanks for adding the tests.

**Pull request merged**